server/tilesource/test.py

```diff
@@ -35,14 +35,66 @@
 
 
 class TestTileSource(TileSource):
-    # TODO: move this class to its own file, and only import if PIL v3+ is available
-    def __init__(self, tileSize=256, levels=10):
+    # TODO: move this class to its own file, and only import if PIL v3+ is
+    # available
+    def __init__(self, tileSize=256, levels=10, params=None):
+        """
+        Initialize the tile class.  The optional params options can include:
+          min: minimum tile level (default 0)
+          max: maximum tile level (default levels - 1)
+          fractal: if 'true', draw a simple fractal on the tiles.
+          w: tile width in pixels (default tileSize)
+          h: tile height in pixels (default tileSize)
+          sizex: image width in pixels at maximum level.
+          sizey: image height in pixels at maximum level.
+
+        :param tileSize: square tile size if not overridden by params.
+        :param levels: number of zoom levels if not overridden by params.  The
+                       allowed zoom levels are [0, levels).
+        :param params: a dictionary of optional parameters.  See above.
+        """
         super(TestTileSource, self).__init__()
-        self.tileSize = tileSize
-        self.levels = levels
-        self.sizeX = (2 ** (self.levels - 1)) * self.tileSize
-        self.sizeY = (2 ** (self.levels - 1)) * self.tileSize
+        self.tileWidth = self.tileHeight = tileSize
+        self.maxlevel = levels - 1
+        self.minlevel = 0
+        self.fractal = False
+        if params:
```

Comment:
My understanding of Girder idioms is that it's the job of the API layer to unpack parameters first.

``minLevel`` (please change to be camelCase), ``fractal``, etc. should all be individual keyword parameters to ``TestTileSource.__init__``. The API then does the ``.get`` and passes them each as arguments directly.

If you were trying to keep ``TestTileSource.__init__`` backwards-comparable, don't worry about it, this plugin is still very early in development. Please feel free to aggressively change any APIs that need it.